### ISSUE TITLE
AI aircraft Losing Labels

### ISSUE THREAD

**Issue by lenvdb** [albar965/littlenavmap#1099]
Dear Albar965,

Running X plane 11 on Little Navmap seems to have problems with AI aircraft labels.
AI aircraft in the air are displaying labels correctly.
I display Registration, Ground speed and Altitude.
No problems with airborne  AI aircraft even when zooming in and out.
But when an AI aircraft takes off from an airfield, only Registration and Ground speed are displayed and not Altitude.
When I zoom out the AI aircraft which has just taken off loses its labels.
The labels are displayed again when I zoom back in.
When I click on that AI aircraft and display its information all the information is there.

Has this something to do with changing  zoom when an aircraft is on the ground?

Thank you for an excellent piece of software.

Len

**Comment by albar965** [comment]
> But when an AI aircraft takes off from an airfield, only Registration and Ground speed are displayed and not Altitude.

Altitude should always be visible when zoomed in. Will check.

> When I zoom out the AI aircraft which has just taken off loses its labels.
> The labels are displayed again when I zoom back in.

This on purpose to avoid cluttering up the map. You can increase the map details to see more labels.

And thank you for the kind words!

**Comment by lenvdb** [comment]
The problem is not with landing aircraft, but when the AI aircraft takes off.
I have also tried X plane 12, which has the same problem.
Also tried Msfs2020 and it works fine.

**Comment by albar965** [comment]
There should be no difference on landing or takeoff for AI regarding labels. I just use the on-ground flags as read from SimConnect plus AGL and speed.

There is also some logic (suggested by a user) to have full labels if AI is near the user aircraft.

**Comment by SandaleAir** [comment]
Hello,

i have the same issue when using Xplane 12 with VATSIM (connected with xPilot). Planes that are departing will not show when airborne when i hide ground traffic. If you zoom in you will see the plane without altitude and speed. Reconnecting xPilot to VATSIM will solve the problem.

Hoping for an solution. ;-)

Greetings,
Sascha.

**Comment by albar965** [comment]
> i have the same issue when using Xplane 12 with VATSIM (connected with xPilot). Planes that are departing will not show when airborne when i hide ground traffic. If you zoom in you will see the plane without altitude and speed. Reconnecting xPilot to VATSIM will solve the problem.

Please post a screenshot of the situation. 

Did you try this setting?
![image](https://github.com/albar965/littlenavmap/assets/15254956/b681c611-fdff-4cf4-916b-69f3a4cc1036)

**Comment by SandaleAir** [comment]
Hello,

i know the setting. I made som screenshots without hiding the aircraft on ground.
![Screenshot 2024-05-06 074745](https://github.com/albar965/littlenavmap/assets/84207216/5bef336f-4bf9-47d0-a425-17dbabd12c17)
The first shows the UAL642 departing from KLAX on ground.
![Screenshot 2024-05-06 075004](https://github.com/albar965/littlenavmap/assets/84207216/67bfe89e-47af-45ae-831e-5ad690b711fe)
After a while the Flag disapears. Befor it didn't show the altitude. It seems Little Navmap did not recognize that the plane is in the air.
![Screenshot 2024-05-06 075028](https://github.com/albar965/littlenavmap/assets/84207216/f2726e3c-1c21-44e4-a06f-64b49c7c4930)
After reconnecting xpilot to VATSIM everything is fine.

Greetings,
Sascha Schuh.

**Comment by albar965** [comment]
Hi Sascha,

you're not connected to VATSIM (LNM options on page `Online Flying`). You get faster aircraft updates but you rely on the aircraft injected by xpilot. I'm suspecting xpilot not giving the correct on-ground status through the simulator since I cannot reproduce this here with my test setup.

Try with VATSIM enabled.

If this does not help:
Keep the VATSIM connection. What happens if you deselect `Remove duplicate aircraft` in options on page `Online Flying`? Also try to disable `View` -> `AI and online aircraft` -> `Show AI and multplayer aircraft` to avoid the aircraft injected into the sim by xpilot.

Best,

Alex

**Comment by SandaleAir** [comment]
Hello Alex,

I also think that it is due to the on ground status of the aircraft that took off. 

If I deactivate remove duplicate aircraft, I see the aircraft twice, once without a flag and once correctly with altitude but with a long time delay. 

Deactivating AI traffic is a very bad option for me because it is important to me to see the current location of the aircraft. Otherwise the time delay is too high. 

Perhaps you could evaluate the rate of climb of the aircraft and then automatically correct the obviously incorrect on ground status in Little Navmap. Or you could use a threshold value for the speed of the aircraft to overwrite this status. But I am not a programmer and I don't know if that works and if it is worth the effort. ;-)

Many thanks for the support,
Sascha Schuh.

**Comment by lenvdb** [comment]
Alex,

I just wanted to add my comment to the issue.
If you look at the aircraft when it looses its labels, you can right click on the AI plane and display its information.
You can then clearly see that all the information is there, such as Call sign, Speed and Altitude.
That must mean that it is all available inside Little Navmap.
But it is not displayed on the chart.

Regards

Len

**Comment by albar965** [comment]
I cannot reproduce this no matter what I'm doing. IVAO, VATSIM, local, AI traffic, all show the labels as they should.

**Comment by lenvdb** [comment]
Alexander,

I set the On line Flying in Tools - Options to none.

And purely rely on data received from X plane 11 / 12, Which does appear in Little Nav Map.
When I query the AI planes, Right click and Show information, it shows me all the information in LNM.

Regards

Len
________________________________

**Comment by albar965** [comment]
Screenshots from XP11 with stock AI. All shown. :man_shrugging: 
![image](https://github.com/albar965/littlenavmap/assets/15254956/75e956c9-f72e-4c22-bd58-11f074d06c5c)
![image](https://github.com/albar965/littlenavmap/assets/15254956/3ade8826-cbe1-4109-803d-5b0fdc46d6cd)

Can you send me a screenshot of the LNM options, page `Map Labels` with the branch `AI, Multiplayer and Online Client Aircraft` expanded?

Alex

**Comment by lenvdb** [comment]
Alexander,

Please find attached pics. as requested.
All is shown using XP12 and Xpilot connected to Vatsim. But same happens with XP11.
AI-1 shows AI aircraft with level 2 detail just taking off.
AI-2 shows aircraft with level 3 detail just taking off, but still not showing Altitude.
If you right click with the mouse you can see all the info for the aircraft is available.

Also shown are my Tool Option settings.

The problem occurs really when AI aircraft are taking off from the airport.

This problem occurs on Xplane 11 / 12.

All is fine on MSFS2020.

Regards

Len
________________________________

**Comment by albar965** [comment]
Sorry, don't see any screenshots.

There is a buffer and a short delay until LNM detects a takeoff. Do not expect altitude values at 50 ft or so.

Apart from that the logic for label visibility is the same for all simulators.

**Comment by albar965** [comment]
This does not work this way. Replying to the Github mail is limited, results in messy posts and does not add images to neither notification emails nor thread answers.
Please reply and add the images in the web interface.

**Comment by lenvdb** [comment]
Here are pics. added in github

![AI-1](https://github.com/albar965/littlenavmap/assets/78753371/52f3024c-9cb0-43c8-ad90-f4efa45e8430)
![AI-2](https://github.com/albar965/littlenavmap/assets/78753371/d156eed2-1eef-4df1-92b7-9505c12d8478)
![Map_labels_AI_aircraft](https://github.com/albar965/littlenavmap/assets/78753371/658fb326-b7f5-4e7a-a03c-d1c68dea58b6)
![Map_labels_user_aircraft](https://github.com/albar965/littlenavmap/assets/78753371/1555ebe1-4325-459c-a25d-6838603d29c8)
![Map_On_line_flying](https://github.com/albar965/littlenavmap/assets/78753371/d09b9f72-c778-4ee4-a0b4-46053d4619fb)

**Comment by albar965** [comment]
Thank you for the screenshots. I'll try again with your combination. Maybe a specific setting triggers this issue.

**Comment by albar965** [comment]
Sorry, cannot reproduce this here.

**Comment by lenvdb** [comment]
Alexander,

Thanks for trying.

There must be something as SandaleAir has the same problems.

It happens both in Xplane 11 and 12, but is fine in MSFS 2020.


Regards

Len
________________________________

**Comment by albar965** [comment]
Just to be sure: Is this X-Plane stock traffic or do you use a traffic add-on?

**Comment by lenvdb** [comment]
Alexander,

I use Xplane 11 steam version.
No addon traffics.
Only Vatsim traffic which comes via X-pilot into the simulator.
Then in Little Navmap only connect to Xplane simulator and not Vatsim.

The data is there , because when I right click on the AI plane all the information is there, but it is just not displayed.
When I log on on and look at AI aircraft in the air all the data is displayed as a label.

Only AI aircraft taking off from the airport loose their labels.

Len

________________________________

**Comment by albar965** [comment]
Thanks for the additional information.

> The data is there , because when I right click on the AI plane all the information is there, but it is just not displayed. When I log on on and look at AI aircraft in the air all the data is displayed as a label.

It does not matter if the data is there or not. If LNM "thinks" that the aircraft is still on ground it will hide most fields even if available. 

Are there any settings in X-Pilot which might affect this?
Why not connect LNM to VATSIM too? The functionality is there.

I would not wonder if this is an issue with X-Pilot. Try it with plain X-Plane AI and no X-Pilot. I tried VATSIM traffic and X-Plane AI and see no issues.

**Comment by lenvdb** [comment]
Alexander,

I have tried it with X-plane AI planes and all works fine.
Also tried with Vatsim, without X-pilot running and again works fine.
The problem with Vatsim is that it only updates every 1-2 minutes.

The problem seems to be associated with X-pilot.
I have checked all the setting, but there is nothing in X-pilot which will change it.

Is there a way to set the labels so that they are always displayed both on the ground and in the air?

Regards

Len
